src/model/level/mod.rs

```diff
@@ -347,16 +382,273 @@ impl Display for Password {
+    pub index_43: Cow<'a, str>,
+
+    /// The amount of objects in this [`PartialLevel`]. Note that a value of `None` _does not_ mean
+    /// that there are no objects in the level, but rather that the server's didn't provide an
+    /// object count.
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `45`, although only for levels uploaded
+    /// in version 2.1 or later. For all older levels this is always `None`
+    pub object_amount: Option<u32>,
+
+    /// According to the GDPS source this is always `1`, although that is
+    /// evidently wrong
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `46` and seems to be an integer
+    pub index_46: Option<Cow<'a, str>>,
+
+    /// According to the GDPS source, this is always `2`, although that is
+    /// evidently wrong
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `47` and seems to be an integer
+    pub index_47: Option<Cow<'a, str>>,
+
+    /// Additional data about this level that can be retrieved by downloading the level.
+    ///
+    /// This is [`None`] for levels retrieved via the "overview" endpoint `getGJLevels`.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub level_data: Option<LevelData<'a>>,
+}
+
+/// Struct encapsulating the additional level data returned when actually downloading a level
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct LevelData<'a> {
+    /// The level's actual data.
+    ///
+    /// TODO: Wrap in fitting Thunk
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `4`, and is urlsafe base64 encoded and `DEFLATE` compressed
+    pub level_data: Cow<'a, str>,
+
+    /// The level's password
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `27`. For encoding details, see the documentation on the
+    /// [`Password`] variants
+    pub password: Password,
+
+    /// The time passed since the `Level` was uploaded, as a string. Note that these strings are
+    /// very imprecise, as they are only of the form "x months ago", or similar.
+    ///
+    /// TODO: Parse these into an enum
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `28`
+    pub time_since_upload: Cow<'a, str>,
+
+    /// The time passed since the `Level` was last updated, as a string. Note that these strings are
+    /// very imprecise, as they are only of the form "x months ago", or similar.
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `29`
+    pub time_since_update: Cow<'a, str>,
+
+    /// According to the GDPS source, this is a value called `extraString`
+    ///
+    /// ## GD Internals:
+    /// This value is provided at index `36`
+    pub index_36: Option<Cow<'a, str>>,
+}
+
 #[cfg(test)]
 mod tests {
     use crate::model::level::{robtop_encode_level_password, Password};
     use base64::URL_SAFE;
 
     #[test]
     fn deserialize_password() {
```

Comment:
Add other types of Passwords tests (NoCopy and FreeCopy)
Do we want to define the behaviour of integers with >6 digits?

Reply:
Sure I can add those tests

I suppose we should just panic in that case, and add some #[should_panic] tests?

Reply:
Kept track of in #13